CRM/Dedupe/Finder.php

```diff
@@ -252,7 +252,7 @@ public static function formatParams($fields, $ctype) {
     // the -digit to civicrm_address.location_type_id and -Primary to civicrm_address.is_primary
     foreach ($flat as $key => $value) {
       $matches = [];
-      if (preg_match('/(.*)-(Primary-[\d+])$|(.*)-(\d+|Primary)$/', $key, $matches)) {
+      if (preg_match('/(.\*)-(Primary-[\d+])$|(.\*)-(\d+-\d+)$|(.\*)-(\d+|Primary)$/', $key, $matches)) {
```

Comment:
I don't understand the escaping of the star character here. What does that accomplish?